frontend/app/src/App.tsx

```diff
@@ -1315,8 +1348,30 @@ export class App extends PureComponent<Props, State> {
   }
 
   onPageChange = (pageScriptHash: string): void => {
-    this.setState({ appLogo: null })
-    this.sendRerunBackMsg(undefined, undefined, pageScriptHash)
+    const { appLogo, elements, mainScriptHash } = this.state
+    // If the app logo is not associated with the main script, clear it.
+    if (appLogo && appLogo.activeScriptHash !== mainScriptHash) {
```

Comment:
is this something we could already handle on the backend? Like, not allowing to send the logo from anywhere else but the main script? I don't have a super strong opinion here, but it feels a little bit weird on first glimpse to have the page_script stuff being tightly bundled with these different areas, such as logo, on the web app side and *think* that it might be cleaner to not send it in the first place. It might also help to detach these concepts from each other from the perspective of using the web app more like a lib - the less complex logic, the better 🙂

Reply:
Ah. I think it should be in the frontend. It's essentially the same logic as the `clearPageNodes`, but the logo is in a special case.

How about logo be in the AppNode as a special variable? That would definitely clean it up.